### ISSUE TITLE
Spell reagents should be consumed on cast completion, not start.

### ISSUE THREAD

**Issue by GeneralGormal** [torilmud/issues#991]
I love the new items that use rare reagents to cast powerful spells, but it REALLY sucks to get knocked down and lose that reagent given how ubiquitous knockdowns are with huge rooms full of bashers, quakes, dragons, etc.  Makes more sense that it would check for the reagent on spell completion anyway.

**Comment by Inames** [comment]
not really, components being consumed in the process of casting seems normal, if your spell is interrupted or fails in some-way partway through, makes total sense that the component could be ruined or destroyed by the failing magic going out of control. imagine the classic fireball expending bat guano to cast, the material is lit and then the resulting blaze is projected toward the enemy, if the spell is interrupted before you throw the ball the material is still burnt, so why should you get to use it again?
